src/CartLatticeExecutor.hpp.Rt

```diff
@@ -0,0 +1,337 @@
+    acc.pop(now);
+    float2 v = now.Color();
+    l = v.x;
+    w = v.y;
+  }
+
+  if (ISFINITE(l)) {
+    l = l * 111;
+    if (               (l <-111)) {r = 255; g = 255; b = 255; }
+    if ((l >= -111) && (l < -11)) {r = 255*(-l-11)/100; g = 255; b = 255; }
+    if ((l >=  -11) && (l <  -1)) {r = 0; g = (255*(-l-1))/10; b = 255; }
+    if ((l >=   -1) && (l <   0)) {r = 0; g = 0; b = 255*(-l); }
+    if ((l >=    0) && (l <   1)) {r = 255*l; g = 0; b = 0; }
+    if ((l >=    1) && (l <  11)) {r = 255; g = 255*(l-1)/10; b = 0; }
+    if ((l >=   11) && (l < 111)) {r = 255; g = 255; b = 255*(l-11)/100; }
+    if ((l >=  111)             ) {r = 255; g = 255; b = 255; }
+    r=r*w;
+    g=g*w + (1-w)*255;
+    b=b*w;
+  } else {
+    r = 255;
+    g = 0;
+    b = 255;
+  }
+  optr[offset].x = r;
+  optr[offset].y = g;
+  optr[offset].z = b;
+  optr[offset].w = 255;
+}
+
+/// Kernel for graphics output
+CudaGlobalFunction void ColorKernel(uchar4 *optr, int z)
+{
+  CartLatticeExecutor::NodeToColor< Primal, NoGlobals, Get >(
+    CudaThread.x+CudaBlock.x*CudaNumberOfThreads.x,
+    CudaBlock.y,
+    z,
+    optr);
+}
+
+/// Runs kernel for rendering graphics
+/**
+  Runs the kernel for rendering graphics
+  \param optr 4-component graphics buffer
+*/
+void CartLatticeExecutor::Color(uchar4 *optr) {
+  CopyToConst();
+  CudaKernelRun(ColorKernel, dim3(floor(m_container.nx / X_BLOCK), m_container.ny, 1), dim3(X_BLOCK), optr, m_container.nz / 2);
+}
+
+// Quantity getters
+<?R
+        for (q in rows(Quantities))
+        {
+                ifdef(q$adjoint); ?>
+/// Calculate quantity [<?%s q$comment ?>] kernel
+/**
+  Kernel to calculate quantity <?%s q$name ?> (<?%s q$comment ?>) over a region
+  \param r Lattice region to calculate the quantity
+  \param tab buffor to put the calculated result
+  \param scale Scale to rescale the result (for units)
+*/
+CudaGlobalFunction void get<?%s q$name ?>(lbRegion r, <?%s q$type ?> * tab, real_t scale)
+{
+  using LA = CartLatticeAccessAll;
+  int x = CudaBlock.x+r.dx;
+  int y = CudaBlock.y+r.dy;
+  int z = CudaBlock.z+r.dz;
+  LA acc(x,y,z); <?R
+    if (q$adjoint) { ?>
+      Node< LA, Adjoint, NoGlobals, Get > now(acc); <?R
+    } else { ?>
+      Node< LA, Primal, NoGlobals, Get > now(acc); <?R
+    }?>
+    <?%s q$type ?> w;
+//  if (now.NodeType) {
+      acc.pop(now); <?R
+      if (q$adjoint) { ?>
+        acc.pop_adj(now); <?R
+      } ?>
+      w = now.get<?%s q$name ?>(); <?R
+      if (q$type == "vector_t") {
+        for (coef in c("x","y","z")) { ?>
+          w.<?%s coef ?> *= scale; <?R
+        }
+      } else { ?>
+        w *= scale; <?R
+      } ?>
+//  } else { <?R
+      if (q$type == "vector_t") {
+        for (coef in c("x","y","z")) { ?>
+//        w.<?%s coef ?> = nan(""); <?R
+        }
+      } else { ?>
+//      w = nan(""); <?R
+      } ?>
+//  }
+  tab[r.offset(x,y,z)] = w;
+}
+
+<?R
+  }
+  ifdef();
+?>
+
+<?R   for (tp in rows(AllKernels)[order(AllKernels$adjoint)]) {
```

Comment:
I would like to have this (template instantiations) in 'cuda.cu'

Reply:
Closed in [375c34e](https://github.com/CFD-GO/TCLB/pull/454/commits/375c34e5f21feba76b06191ad6c07c2c20fa4da0)